Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/FileTransporter.cs

```diff
@@ -0,0 +1,495 @@
+                        }
+
+                        // Check does the destination directory exists.
+                        if (!client.Exists(_batchContext.Destination.Directory))
+                        {
+                            if (_batchContext.Options.CreateDestinationDirectories)
+                            {
+                                try
+                                {
+                                    CreateAllDirectories(client, _batchContext.Destination.Directory);
+                                }
+                                catch (Exception ex)
+                                {
+                                    userResultMessage = $"Error while creating destination directory '{_batchContext.Destination.Directory}': {ex.Message}";
+                                    return FormFailedFileTransferResult(userResultMessage);
+                                }
+                            }
+                            else
+                            {
+                                userResultMessage = $"Destination directory '{_batchContext.Destination.Directory}' was not found.";
+                                return FormFailedFileTransferResult(userResultMessage);
+                            }
+                        }
+
+                        client.ChangeDirectory(_batchContext.Destination.Directory);
+
+                        _batchContext.DestinationFiles = client.ListDirectory(".");
+
+                        foreach (var file in files)
+                        {
+                            var singleTransfer = new SingleFileTransfer(file, _batchContext, client, _renamingPolicy, _logger);
+                            var result = singleTransfer.TransferSingleFile();
+                            _result.Add(result);
+                        }
+                        client.Disconnect();
+                    }
+                }   
+            }
+            catch (SshConnectionException ex)
+            {
+                userResultMessage = $"Error when establishing connection to the Server: {ex.Message}, {userResultMessage}";
+                return FormFailedFileTransferResult(userResultMessage);
+            }
+            catch (SocketException)
+            {
+                userResultMessage = $"Unable to establish the socket: No such host is known. {userResultMessage}";
+                return FormFailedFileTransferResult(userResultMessage);
+            }
+            catch (SshAuthenticationException ex)
+            {
+                userResultMessage = $"Authentication of SSH session failed: {ex.Message}, {userResultMessage}";
+                return FormFailedFileTransferResult(userResultMessage);
+            }
+
+            return FormResultFromSingleTransferResults(_result);
+        }
+
+        #region Helper methods
+        /// <summary>
+        /// Form connection info for the SftpClient class.
```

Comment:
I am pretty sure everyone will understand the method without the comment

Reply:
Comment removed.